hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/node/TestNodeStateManager.java

```diff
@@ -202,6 +202,58 @@ public void testNodeOpStateCanBeSet()
         newStatus.getHealth());
   }
 
+  @Test
+  public void testHealthEventsFiredWhenOpStateChanged()
+      throws NodeAlreadyExistsException, NodeNotFoundException {
+    DatanodeDetails dn = generateDatanode();
+    nsm.addNode(dn);
+
+    // First set the node to decommissioned, then run through all op states in
+    // order and ensure the non_healthy_to_healthy event gets fired
+    nsm.setNodeOperationalState(dn,
+        HddsProtos.NodeOperationalState.DECOMMISSIONED);
+    for (HddsProtos.NodeOperationalState s :
+        HddsProtos.NodeOperationalState.values()) {
+      eventPublisher.clearEvents();
+      nsm.setNodeOperationalState(dn, s);
+      assertEquals("NON_HEALTHY_TO_HEALTHY_NODE",
+          eventPublisher.getLastEvent().getName());
```

Comment:
I think it would be safer to compare the event, not its name:

```
      assertEquals(SCMEvents.NON_HEALTHY_TO_HEALTHY_NODE,
          eventPublisher.getLastEvent());
```

Reply:
True. I was pretty much copying existing code when I added this test. I have made this change here and also applied it to the other occurrences in the class to keep in consistent.